compiler/optimizer/OMRSimplifierHelpers.cpp

```diff
@@ -602,60 +602,47 @@ TR::Node *reduceExpTwoAndGreaterToMultiplication(int32_t exponentValue, TR::Node
+
+   TR::Node *resultNode = NULL;
+
+   // There are two algorithms here -- they are equivalent in the number 
+   // of multiply operations however the second is better for platforms 
+   // that have a destructive multiply instruction as less clobber evaluates 
+   // will be required. The second has the advantage that more parallel 
+   // multiply operations are created
    if (s->comp()->cg()->multiplyIsDestructive())
       {
-      TR::Node * node = NULL;
       int32_t bitPosOfLeftMostOne = 32 - leadingZeroes(exponentValue) - 1; // bitPos=0 is the least significant bit so for exponentValue=7 : 32 - 29 - 1 = 2
-      node = baseNode;
+      resultNode = baseNode;
       if (bitPosOfLeftMostOne != 0)
          {
          for (int32_t i = bitPosOfLeftMostOne-1; i >= 0; i--)
             {
-            node = TR::Node::create(multOp, 2, node, node);
-#ifdef J9_PROJECT_SPECIFIC
-            if (isPacked)
-               node->setPDMulPrecision();
-#endif
+            resultNode = TR::Node::create(multOp, 2, resultNode, resultNode);
             dumpOptDetails(s->comp(), "%screated %s [" POINTER_PRINTF_FORMAT "] operation for exponentiation strength reduction (algorithmA/caseA)\n",
-                             s->optDetailString(), node->getOpCode().getName(), node);
+                             s->optDetailString(), resultNode->getOpCode().getName(), resultNode);
             if (((exponentValue >> i)&0x1) != 0)
                {
-               node = TR::Node::create(multOp, 2, node, baseNode);
-#ifdef J9_PROJECT_SPECIFIC
-               if (isPacked)
-                  node->setPDMulPrecision();
-#endif
+               resultNode = TR::Node::create(multOp, 2, resultNode, baseNode);
                dumpOptDetails(s->comp(), "%screated %s [" POINTER_PRINTF_FORMAT "] operation for exponentiation strength reduction (algorithmA/caseB)\n",
-                                s->optDetailString(), node->getOpCode().getName(), node);
+                                s->optDetailString(), resultNode->getOpCode().getName(), resultNode);
                }
             }
          }
-      return node;
       }
    else
       {
       int32_t maxCeiling = ceilingPowerOfTwo(maxExponent);  // if maxExponent=31 then maxCeiling = 32
       int32_t maxCeilingExp = trailingZeroes(maxCeiling);   // (2^x=maxCeiling) so if maxCeiling = 32 then x=maxCeilingExp=5
-      TR::Node * node = NULL;
-      TR::Node ** subTrees = (TR::Node**)s->comp()->trMemory()->allocateStackMemory((maxCeilingExp+1)*sizeof(TR::Node*));
+      TR::Node **subTrees = (TR::Node**)s->comp()->trMemory()->allocateStackMemory((maxCeilingExp+1)*sizeof(TR::Node*));
       subTrees[0] = baseNode;
       int32_t i = 0;
       for (i = 1; exponentValue >= (CONSTANT64(1) << i); ++i) // i can reach maxCeilingExp+1
          {
          int32_t j = i-1;
-         node = subTrees[i] = TR::Node::create(multOp, 2, subTrees[j], subTrees[j]);
-#ifdef J9_PROJECT_SPECIFIC
```

Comment:
Yes, the recursive method setNodePrecisionIfNeeded for the node is called at the end of the method before returning the node.